src/main.lua

```diff
@@ -471,26 +517,28 @@ Handlers.add(ActionMap.BuyRecord, utils.hasMatchingTag("Action", ActionMap.BuyRe
 		msg.Timestamp,
 		msg.Tags["Process-Id"]
 	)
-	if not status then
-		ao.send({
-			Target = msg.From,
-			Tags = {
-				Action = "Invalid-Buy-Record-Notice",
-				Error = "Invalid-Buy-Record",
-			},
-			Data = tostring(result),
-		})
+	if not shouldContinue2 then
 		return
-	else
-		ao.send({
-			Target = msg.From,
-			Tags = { Action = "Buy-Record-Notice", Name = msg.Tags.Name },
-			Data = json.encode(result),
-		})
 	end
+
+	local record = {}
+	if result ~= nil then
+		record = result.record
+		addRecordResultFields(ioEvent, result)
+	end
+
+	ao.send({
+		Target = msg.From,
+		Tags = { Action = "Buy-Record-Notice", Name = msg.Tags.Name },
+		Data = json.encode(record),
+	})
+	ioEvent:printEvent()
```

Comment:
i don't think it's supported, but it would be nice if we could add a `postpend` handler that always prints this IO event before returning.

Reply:
Was thinking the same...